service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ImmunizationObservationStatementMapper.java

```diff
@@ -223,4 +228,27 @@ private String extractVaccinationProtocolString(Immunization.ImmunizationVaccina
 
         return vaccinationProtocol + VACCINATION_TARGET_DISEASE + targetDiseases;
     }
+
+    private String buildCode(Immunization immunization) {
+        var vaccinationProcedure = ExtensionMappingUtils.filterExtensionByUrl(immunization, VACCINATION_PROCEDURE_URL);
+        if (vaccinationProcedure.isPresent()) {
+            CodeableConcept codeableConcept = (CodeableConcept) vaccinationProcedure.get().getValue();
+            return codeableConceptCdMapper.mapCodeableConceptToCd(codeableConcept);
+        }
+        return StringUtils.EMPTY;
+    }
+
+    private String buildVaccineCode(Immunization immunization) {
+        if (immunization.hasVaccineCode() && vaccineCodeNotUNK(immunization.getVaccineCode())) {
+            var code = CodeableConceptMappingUtils.extractTextOrCoding(immunization.getVaccineCode());
+            if (code.isPresent()) {
+                return String.format(VACCINATION_CODE, code.get());
+            }
+        }
+        return StringUtils.EMPTY;
+    }
+
+    private boolean vaccineCodeNotUNK(CodeableConcept codeableConcept) {
```

Comment:
The vaccinationProcedure extension is required. Should throw error is missing. Return empty only for UNK code.